include/envoy/http/codec.h

```diff
@@ -132,6 +142,14 @@ class Stream {
    * @param reason supplies the reset reason.
    */
   virtual void resetStream(StreamResetReason reason) PURE;
+
+  /**
+   * Enable/disable further data from this stream.
+   * Cessation of data may not be immediate.  For example, for HTTP/2 this may stop further flow
+   * control window updates which will result in the peer eventually stopping sending data.
```

Comment:
> everything seemlessly avoids the double buffering

I assume you mean by rejiggering everything so that encode* returns the number of bytes accepted, etc.? That's certainly doable, though, buffering still will need to happen since whoever fails to write will need to buffer locally until they can write. This pushes more complexity out to the leaves. Not sure it's worth it, but I guess we can see how things go with this approach.